core/lib/instrument.c

```diff
@@ -767,6 +767,14 @@ void free_all_callback_lists()
 }
 
 void
+instrument_delete_locks(void)
```

Comment:
The name is misleading: most lock deletion is still in instrument_exit().  Maybe instrument_exit_post_sideline() or sthg is better?